src/pages/posts.xml.ts

```diff
@@ -0,0 +1,20 @@
+import rss from "@astrojs/rss";
+import { getCollection } from "astro:content";
+import type { APIContext } from "astro";
+
+export async function GET(context: APIContext) {
```

Comment:
How can we test this

1) locally/manually, in a bun dev environment and 
2) in test files

To make sure that our RSS feed includes everything it should and nothing it shouldn't?

Reply:
1. Testing locally is pretty straightforward, just navigate to `/posts.xml`, and you should see the XML RSS feed, and can visually match the items to the files in the `src/content/pages` directory.
2. This is more difficult to test. My only thought to thoroughly verify that the RSS matches would be parsing the XML returned in the body of the endpoint, and validating that that contains all of the attributes it needs. That comes with a bit of overhead though, so open to suggestions before implementing it this way.

Reply:
The existence of the collection being parsed is handled with the md test suite, really just validating that the XML file exists and contains the correct number of tags (matching collection length) would be the best way to go about it

Reply:
In the test runner DO NOT use get collection as if it fails the runner will pass with a forEach since forEach of none is an auto pass, instead do what is done in all other tests and raw parse the dir with `fs`